spec/deep_structure_copy_spec.rb

```diff
@@ -129,6 +129,20 @@ class CopyableWarranty < ActiveRecord::Base
         expect(CopyableVehicle.count).to eq(2)
         expect(CopyableAmenity.count).to eq(2)
       end
+
+      context 'with a global override, can copy vehicle with new owner' do
+        before(:each) do
+          @jane = CopyableOwner.create!(name: 'Jane')
+        end
+
+        it 'should copy the records correctly' do
+          @copy_of_joes_car = @porsche.create_copy!(global_override: { copyable_owner_id: @jane.id })
+          expect(CopyableVehicle.count).to eq(2)
+          expect(CopyableAmenity.count).to eq(2)
+          expect(@copy_of_joes_car.copyable_owner_id).to eq(@jane.id)
+          expect(@copy_of_joes_car.copyable_amenities.first.copyable_owner_id).to eq(@jane.id)
```

Comment:
You could map over all the amenities and see if they all get the right `copyable_owner_id`, couldn't you?

Reply:
I could, I didn't realize there would be more than one. Will do!